protocols/migration/migration_reports_5_0_0_to_reports_6_0_0.py

```diff
@@ -25,3 +25,172 @@ def migrate_interpretation_request_rd(self, old_instance):
+        return self.validate_object(
+            object_to_validate=new_variant, object_type=self.new_model.SmallVariant
+        )
+
+    def migrate_variant_attributes(self, old_variant):
+        # This logic is done here because alleleOrigins is part of the
+        # v6 variantAttributes, which is non-nullable in v5 (or v6)
+        # so the v5 variantAttributes could be null, but the alleleOrigins can not be,
+        # so this is done to make sure we don't lose the alleleOrigins
+        if old_variant.variantAttributes is not None:
+            new_variant_attributes = self.convert_class(self.new_model.VariantAttributes, old_variant.variantAttributes)
+        else:
+            new_variant_attributes = self.new_model.VariantAttributes()
+
+        new_variant_attributes.genomicChanges = old_variant.genomicChanges
+        new_variant_attributes.cdnaChanges = old_variant.cdnaChanges
+        new_variant_attributes.proteinChanges = old_variant.proteinChanges
+        new_variant_attributes.additionalTextualVariantAnnotations = old_variant.additionalTextualVariantAnnotations
+        new_variant_attributes.references = old_variant.references
+        new_variant_attributes.additionalNumericVariantAnnotations = old_variant.additionalNumericVariantAnnotations
+        new_variant_attributes.comments = old_variant.comments
+        new_variant_attributes.alleleOrigins = self.migrate_allele_origins(old_origins=old_variant.alleleOrigins)
+        new_variant_attributes.alleleFrequencies = self.migrate_allele_frequencies(old_frequencies=old_variant.alleleFrequencies)
+        variant_identifiers = self.new_model.VariantIdentifiers(
+            dbSnpId=old_variant.dbSnpId,
+            cosmicIds=old_variant.cosmicIds,
+            clinVarIds=old_variant.clinVarIds,
+        )
+        variant_identifiers = self.validate_object(object_to_validate=variant_identifiers, object_type=self.new_model.VariantIdentifiers)
+        new_variant_attributes.variantIdentifiers = variant_identifiers
+        return self.validate_object(object_to_validate=new_variant_attributes, object_type=self.new_model.VariantAttributes)
+
+    def migrate_allele_frequencies(self, old_frequencies):
+        if old_frequencies is None:
+            return None
+        return [self.migrate_allele_frequency(old_frequency=old_frequency) for old_frequency in old_frequencies]
+
+    def migrate_allele_frequency(self, old_frequency):
+        new_frequency = self.convert_class(self.new_model.AlleleFrequency, old_frequency)
+        return self.validate_object(object_to_validate=new_frequency, object_type=self.new_model.AlleleFrequency)
+
+    def migrate_allele_origins(self, old_origins):
+        return [self.migrate_allele_origin(origin=origin) for origin in old_origins]
+
+    def migrate_allele_origin(self, origin):
+        origin_map = {
+            self.old_model.AlleleOrigin.de_novo_variant: self.new_model.AlleleOrigin.de_novo_variant,
```

Comment:
why do we need to map these fields? I thought these would be filled by `self.convert_class()` as they are named the same.

Reply:
removed